File: pset6/cash.py
from cs50 import get_float
from math import ceil

def main():
    cash = 0
    coins = 0
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1

    while cash <= 0:
        cash = get_float("Change owed: ")

    cash = 100*cash

    if cash >= quarter:
        coins += cash // quarter
        cash = ceil(cash % quarter)

    if cash >= dime:
        coins += cash // dime
        cash = ceil(cash % dime)

    if cash >= nickel:
        coins += cash // nickel
        cash = ceil(cash % nickel)

    if cash >= penny:
        coins += cash // penny
        cash = ceil(cash % penny)

    print(f"{int(coins)}")

if __name__ == '__main__':
    main()

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main()
{
    int quarter = 25, dime = 10, nickel = 5, penny = 1, sif;
    float change=-1;
    
    while(change<0)
    {
        change = get_float("Change owed: ");
    }
    sif = round(change*100); // Sif is my dog's name. Therefore, it'll always be the main variable.
    
    int coins = -1;
    
    while(coins<0)
    {
        coins++;
        if(sif>=25)
        {
            coins += sif/quarter;
            sif = sif%quarter;
        }
        if(sif>=10)
        {
            coins += sif/dime;
            sif = sif%dime;
        }
        if(sif>=5)
        {
            coins += sif/nickel;
            sif = sif%nickel;
        }
        coins += sif/penny;
    }
    printf("%i\n", coins);
}
